frontend/src/components/Budget/TransactionForm.tsx

```diff
@@ -1,37 +1,270 @@
 import * as React from 'react';
-// import * as R from 'ramda';
-import TransactionFormItem from './TransactionFormItem';
+import { connect } from 'react-redux';
+import Select from 'react-select';
+import {
+  Formik,
+  Form,
+  Field,
+  FieldArray,
+  FormikProps,
+  FieldArrayRenderProps,
+  FormikValues,
+  FieldProps,
+} from 'formik';
+import * as Yup from 'yup';
+import {
+  TransactionRequest,
+  Category,
+  TransactionFormItem,
+} from 'src/types/budget';
+import { AppState, APIProps } from 'src/types';
+import * as R from 'ramda';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import { saveTransactions } from 'src/actions/budget';
+import { createLoadingSelector } from 'src/selectors';
 
 interface Props {
-  transactions: any;
+  categories: Category[];
+  saveTransactions: (transactions: TransactionRequest[]) => void;
 }
 
-interface State {
-  count: number;
-  editing: boolean;
-  temp: string;
+const transactionsValidationSchema = Yup.object().shape({
+  transactions: Yup.array().of(
+    Yup.object().shape({
+      categoryId: Yup.number().required(),
+      type: Yup.string()
+        .oneOf(['deposit', 'withdraw'])
+        .required(),
+      amount: Yup.number()
+        .positive('This field must be positive number')
+        .required(),
+      description: Yup.string(),
+      transactionDatetime: Yup.date().required(),
+    })
+  ),
+});
+
+const generateTransaction = (categoryId: number): TransactionFormItem => ({
+  categoryId,
+  description: '',
+  type: 'deposit',
+  amount: 0,
+  transactionDatetime: new Date(),
+});
+
+interface CategoryOption {
+  value: Category;
+  label: string;
 }
 
-export default class TransactionForm extends React.Component<Props, State> {
-  addTransactionFormItem(e) {
-    e.preventDefault();
-  }
+class TransactionForm extends React.Component<Props & APIProps> {
+  handleSubmit = (values: FormikValues) => {
+    const transactions = values.transactions.map(
+      (value: TransactionFormItem): TransactionRequest => {
+        const {
+          categoryId,
+          type,
+          amount,
+          description,
+          transactionDatetime,
+        } = value;
+        return {
+          categoryId,
+          withdraw: type === 'withdraw' ? amount : 0,
```

Comment:
I think BE validation was whether `withdraw` / `deposit` has an amount or is null.
IIRC, BE is not even allowing both `withdraw` and `deposit` to be 0 meaning one of them has to be null.